lang/funcs/core/sys/uptime_fact.go

```diff
@@ -0,0 +1,64 @@
+package coresys
+
+import (
+	"time"
+
+	"github.com/purpleidea/mgmt/lang/funcs/facts"
+	"github.com/purpleidea/mgmt/lang/types"
+
+	errwrap "github.com/pkg/errors"
+)
+
+func init() {
+	facts.ModuleRegister(moduleName, "uptime", func() facts.Fact { return &UptimeFact{} })
+}
+
+type UptimeFact struct {
+	init      *facts.Init
+	closeChan chan struct{}
+}
+
+func (obj *UptimeFact) Info() *facts.Info {
+	return &facts.Info{
+		Output: types.NewType("int"),
```

Comment:
There's a shortcut for the "core" primitives: https://github.com/purpleidea/mgmt/blob/4c8086977a5bbd0ffb3c991d9fdc093092f369bf/lang/types/type.go#L32
Please use that instead.